File: prettier-plugin-csharpier/src/Printer/Types/ExpressionStatement.ts
import { Doc } from "prettier";
import { printTrailingComments } from "../PrintComments";
import { PrintMethod } from "../PrintMethod";
import { SyntaxToken, SyntaxTreeNode } from "../SyntaxTreeNode";
import { concat, group, hardline, indent, join, softline, line, doubleHardline } from "../Builders";

export interface ExpressionStatementNode extends SyntaxTreeNode<"ExpressionStatement"> {
    expression: SyntaxTreeNode;
    semicolonToken: SyntaxToken;
}

export const printExpressionStatement: PrintMethod<ExpressionStatementNode> = (path, options, print) => {
    const node = path.getValue();
    const parts: Doc[] = [path.call(print, "expression"), ";"];
    printTrailingComments(node, parts, "semicolonToken");
    return concat(parts);
};

File: Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/ExpressionStatement.cs
namespace CSharpier.SyntaxPrinter.SyntaxNodePrinters;

internal static class ExpressionStatement
{
    public static Doc Print(ExpressionStatementSyntax node, PrintingContext context)
    {
        return Doc.Group(
            ExtraNewLines.Print(node),
            Node.Print(node.Expression, context),
            Token.Print(node.SemicolonToken, context)
        );
    }
}
